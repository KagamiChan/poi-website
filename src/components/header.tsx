import { faLanguage } from '@fortawesome/free-solid-svg-icons/faLanguage'
import { faSwatchbook } from '@fortawesome/free-solid-svg-icons/faSwatchbook'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import map from 'lodash/map'
import { CommandBarButton } from 'office-ui-fabric-react'
import React, { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

import { getExactLanguage } from '../utils'

export const languages = {
  en: 'English',
  ja: '日本語',
  'zh-Hans': '简体中文',
  'zh-Hant': '繁體中文',
}

const Container = styled.div`
  height: 60px;

  background-color: white;
  border-bottom: 1px solid #eee;
`

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: 960px;
  display: flex;
  font-size: 20px;
`

const LinkItem = styled.a`
  padding: 0 1ex;
  line-height: 60px;
  display: inline-block;
  cursor: pointer;
  transition: 0.3s;
  position: relative;

  display: none;

  :hover {
    ::before {
      content: '';
      display: block;
      background-color: #333;
      position: absolute;
      height: 2px;
      width: 100%;
      bottom: 0;
      left: 0;
    }
  }
`

const Spacer = styled.div`
  flex: 1;
`

const CommandBar = styled.div`
  display: flex;
  align-items: center;
  height: 60px;
  line-height: 60px;

  button {
    height: 60px;
    padding: 0 1ex;
  }
`

const Icon = styled(FontAwesomeIcon)`
  margin-right: 1ex;
`

interface IProps {
  onChangeTheme: () => void
  isDark: boolean
}

export const Header = ({ onChangeTheme, isDark }: IProps) => {
  const { t, i18n } = useTranslation()

  const lang = getExactLanguage(i18n.language)

  const options = useMemo(
    () =>
      map(languages, (value, key) => ({
        key,
        onClick: () => {
          i18n.changeLanguage(key)
        },
        text: value,
      })),
    [i18n.changeLanguage],
  )

  return (
    <Container>
      <Wrapper>
        <div>
          <LinkItem>{t('Explore')}</LinkItem>
          <LinkItem>{t('Plugins')}</LinkItem>
        </div>
        <Spacer />
        <CommandBar>
          <CommandBarButton onClick={onChangeTheme}>
            <Icon icon={faSwatchbook} />
            {t('current-theme')}
            {t(isDark ? 'Chibaheit' : 'Lilywhite')}
          </CommandBarButton>
          <CommandBarButton
            menuProps={{
              items: options,
            }}
            data-testid="language-dropdown"
          >
            <Icon icon={faLanguage} />
            {t('language')}
          </CommandBarButton>
        </CommandBar>
      </Wrapper>
    </Container>
  )
}
