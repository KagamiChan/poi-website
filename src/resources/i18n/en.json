{
  "language": "English",
  "name": "poi",
  "description": "Scalable KanColle browser and tool.",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "Help us via OpenCollective",
  "weibo": "Weibo",
  "download-for": "Download for",
  "linux-x64": "Linux (x64)",
  "linux-deb-x64": "Debian / Ubuntu (x64)",
  "linux-rpm-x64": "Fedora / CentOS (x64)",
  "macos-x64": "macOS (x64)",
  "win-ia32": "Windows (x86) Portable",
  "win-ia32-setup": "Windows (x86)",
  "win-x64": "Windows (x64) Portable",
  "win-x64-setup": "Windows (x64)",
  "stable-hint": "Stable, for most users",
  "beta-hint": "Beta, for new features",
  "other-versions": "Check the complete list",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord sub-channel"
}
