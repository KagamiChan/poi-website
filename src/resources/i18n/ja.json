{
  "language": "日本語",
  "name": "poi",
  "description": "拡張可能な艦隊これくしょんブラウザ",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "Help us via OpenCollective",
  "weibo": "Weibo",
  "download-for": "ダウンロード: ",
  "linux-x64": "64 ビット Linux バージョン",
  "linux-deb-x64": "64 ビット Debian / Ubuntu バージョン",
  "linux-rpm-x64": "64 ビット Fedora / CentOS バージョン",
  "macos-x64": "64 ビット macOS バージョン",
  "win-ia32": "32 ビット Windows バージョン",
  "win-ia32-setup": "32 ビット Windows インストーラ",
  "win-x64": "64 ビット Windows バージョン",
  "win-x64-setup": "64 ビット Windows インストーラ",
  "stable-hint": "安定版・一般利用の方におすすめ",
  "beta-hint": "開発版・新機能が欲しい方におすすめ",
  "other-versions": "他のバージョン",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord チャンネル"
}
