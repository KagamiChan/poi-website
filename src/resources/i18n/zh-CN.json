{
  "language": "简体中文",
  "name": "poi",
  "description": "一个可扩展的舰队 Collection 浏览器",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "通过 OpenCollective 帮助我们",
  "weibo": "新浪微博",
  "download-for": "下载",
  "linux-x64": "64 位 Linux 版",
  "linux-deb-x64": "64 位 Debian / Ubuntu 版",
  "linux-rpm-x64": "64 位 Fedora / CentOS 版",
  "macos-x64": "64 位 macOS 版",
  "win-ia32": "32 位 Windows 版",
  "win-ia32-setup": "32 位 Windows 安装包",
  "win-x64": "64 位 Windows 版",
  "win-x64-setup": "64 位 Windows 安装包",
  "stable-hint": "正式版，适合大部分人使用",
  "beta-hint": "测试版，包含最新的功能",
  "other-versions": "下载其他版本",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord 频道"
}
