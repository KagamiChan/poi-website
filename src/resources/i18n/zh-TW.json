{
  "language": "正體中文",
  "name": "poi",
  "description": "一個可擴展的艦隊 Collection 瀏覽器",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "Help us via OpenCollective",
  "weibo": "新浪微博",
  "download-for": "下載",
  "linux-x64": "64 位 Linux 版",
  "linux-deb-x64": "64 位 Debian / Ubuntu 版",
  "linux-rpm-x64": "64 位 Fedora / CentOS 版",
  "macos-x64": "64 位 macOS 版",
  "win-ia32": "32 位 Windows 版",
  "win-ia32-setup": "32 位 Windows 安裝包",
  "win-x64": "64 位 Windows 版",
  "win-x64-setup": "64 位 Windows 安裝包",
  "stable-hint": "正式版，適合大部分人使用",
  "beta-hint": "測試版，包含最新的功能",
  "other-versions": "瀏覽完整列表",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord 頻道",
  "Choose another platform": "選擇其它平臺",
  "Explore": "功能簡介",
  "Plugins": "擴展列表"
}
