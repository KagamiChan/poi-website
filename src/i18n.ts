import i18next from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { initReactI18next } from 'react-i18next'

import en from './resources/i18n/en.json'
import ja from './resources/i18n/ja.json'
import zhCN from './resources/i18n/zh-CN.json'
import zhTW from './resources/i18n/zh-TW.json'

const resources = {
  en: { translation: en },
  ja: { translation: ja },
  'zh-CN': { translation: zhCN },
  'zh-TW': { translation: zhTW },
}

export const i18n = i18next.createInstance()

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: process.env.NODE_ENV === 'development',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false,
    },
    resources,
  })
