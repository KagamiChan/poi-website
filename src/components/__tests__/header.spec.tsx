import { getByText } from '@testing-library/dom'
import { act, fireEvent, render } from '@testing-library/react'
import _ from 'lodash'
import React from 'react'

import { i18n } from '../../i18n'
import { Header, languages } from '../header'

const noop = () => {
  /* */
}

describe('<Header />', () => {
  it('renders', () => {
    const { asFragment, getByTestId, baseElement } = render(
      <Header isDark={true} onChangeTheme={noop} />,
    )
    const dropdown = getByTestId('language-dropdown')
    fireEvent.click(dropdown)
    const layer = baseElement.querySelector('.ms-Layer')
    expect(layer).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    expect(layer).toMatchSnapshot()
  })

  it('changes language', () => {
    const { getByTestId, asFragment, baseElement, debug } = render(
      <Header isDark={true} onChangeTheme={noop} />,
    )
    const origin = asFragment()
    _.each(languages, (value, name) => {
      const dropdown = getByTestId('language-dropdown')
      fireEvent.click(dropdown)
      const layer = baseElement.querySelector('.ms-ContextualMenu-list.is-open')
      expect(layer).toBeTruthy()
      fireEvent.click(getByText(layer as HTMLElement, value))
      expect(i18n.language).toBe(name)
      expect(origin).toMatchDiffSnapshot(asFragment(), {}, name)
    })

    _.each(languages, async (value, name) => {
      act(() => {
        i18n.changeLanguage(name)
      })
      expect(origin).toMatchDiffSnapshot(asFragment(), {}, name)
    })
  })
})
