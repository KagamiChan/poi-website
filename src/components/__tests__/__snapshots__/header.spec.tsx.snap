// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> changes language: en 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`<Header /> changes language: en 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`<Header /> changes language: ja 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           機能紹介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           プラグイン一覧
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,17 +38,17 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"日本語\\"
          >
            日本語
          </a>
          <a"
`;

exports[`<Header /> changes language: ja 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           機能紹介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           プラグイン一覧
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,17 +38,17 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"日本語\\"
          >
            日本語
          </a>
          <a"
`;

exports[`<Header /> changes language: zh-Hans 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           功能简介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           插件列表
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,11 +38,11 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
@@ -50,11 +50,11 @@
            title=\\"日本語\\"
          >
            日本語
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"简体中文\\"
          >
            简体中文
          </a>
          <a"
`;

exports[`<Header /> changes language: zh-Hans 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           功能简介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           插件列表
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,11 +38,11 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
@@ -50,11 +50,11 @@
            title=\\"日本語\\"
          >
            日本語
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"简体中文\\"
          >
            简体中文
          </a>
          <a"
`;

exports[`<Header /> changes language: zh-Hant 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           功能簡介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           擴展列表
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,11 +38,11 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
@@ -56,11 +56,11 @@
            title=\\"简体中文\\"
          >
            简体中文
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"繁體中文\\"
          >
            繁體中文
          </a>
        </div>"
`;

exports[`<Header /> changes language: zh-Hant 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,16 +7,16 @@
      >
        <div>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Explore
+           功能簡介
          </a>
          <a
            class=\\"sc-htpNat jwNqkS\\"
          >
-           Plugins
+           擴展列表
          </a>
        </div>
        <div
          class=\\"sc-bxivhb hYiooe\\"
        />
@@ -38,11 +38,11 @@
              d=\\"M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z\\"
              fill=\\"currentColor\\"
            />
          </svg>
          <a
-           class=\\"sc-EHOje bOTOim\\"
+           class=\\"sc-EHOje hgcoFM\\"
            title=\\"English\\"
          >
            English
          </a>
          <a
@@ -56,11 +56,11 @@
            title=\\"简体中文\\"
          >
            简体中文
          </a>
          <a
-           class=\\"sc-EHOje hgcoFM\\"
+           class=\\"sc-EHOje bOTOim\\"
            title=\\"繁體中文\\"
          >
            繁體中文
          </a>
        </div>"
`;

exports[`<Header /> renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa kFhlbX"
  >
    <div
      class="sc-bwzfXH hYWPlG"
    >
      <div>
        <a
          class="sc-htpNat jwNqkS"
        >
          Explore
        </a>
        <a
          class="sc-htpNat jwNqkS"
        >
          Plugins
        </a>
      </div>
      <div
        class="sc-bxivhb hYiooe"
      />
      <div
        class="sc-ifAKCX fsCAIN"
        data-testid="language-list"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-language fa-w-20 "
          data-icon="language"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z"
            fill="currentColor"
          />
        </svg>
        <a
          class="sc-EHOje bOTOim"
          title="English"
        >
          English
        </a>
        <a
          class="sc-EHOje hgcoFM"
          title="日本語"
        >
          日本語
        </a>
        <a
          class="sc-EHOje hgcoFM"
          title="简体中文"
        >
          简体中文
        </a>
        <a
          class="sc-EHOje hgcoFM"
          title="繁體中文"
        >
          繁體中文
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
